File: leetcode/leetcode_py/greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1: return ""
        def findGCD(a, b):
            return a if b == 0 else findGCD(b, a%b) # or :math.gcd(a, b)
        return str1[:findGCD(len(str1), len(str2))]

File: leetcode/leetcode_cpp/greatest-common-divisor-of-strings.cpp
class Solution {
public:
    int findGCD(int a, int b) {
        if (b == 0) return a;
        return findGCD(b, a%b);
    }
    string gcdOfStrings_1(string& str1, string& str2) {
        if (str1 + str2 != str2 + str1) return "";
        return str1.substr(0, findGCD(str1.size(), str2.size()));
    }
    string gcdOfStrings_2(string& str1, string& str2) {
        if (str1 + str2 != str2 + str1) return "";
        return str1.substr(0, gcd(str1.size(), str2.size()));
    }
    string gcdOfStrings(string str1, string str2) {
        return gcdOfStrings_1(str1, str2);
    }
};